chatkit-core/src/main/kotlin/com/pusher/chatkit/SynchronousChatManager.kt

```diff
@@ -198,31 +194,38 @@ class SynchronousChatManager constructor(
     }
 
     private fun transformCursorsSubscriptionEvent(event: CursorSubscriptionEvent): ChatEvent =
-                when (event) {
-                    is CursorSubscriptionEvent.OnCursorSet ->
-                        ChatEvent.NewReadCursor(event.cursor)
-                    else ->
-                        ChatEvent.NoEvent
-                }
+            when (event) {
+                is CursorSubscriptionEvent.OnCursorSet -> ChatEvent.NewReadCursor(event.cursor)
+                else -> ChatEvent.NoEvent
+            }
 
     private fun applyUserSubscriptionEvent(event: UserSubscriptionEvent): List<UserSubscriptionEvent> =
         when (event) {
             is UserSubscriptionEvent.InitialState -> {
-                val removedFrom = (roomService.roomStore.toList() - event.rooms).also {
+                val knownRooms = roomService.roomStore.toList()
+                val removedFrom = (knownRooms - event.rooms).also {
                     roomService.roomStore -= it
                 }.map {
                     UserSubscriptionEvent.RemovedFromRoomEvent(it.id)
                 }
 
-                val addedTo = (event.rooms - roomService.roomStore.toList()).also {
+                val addedTo = (event.rooms - knownRooms).also {
                     roomService.roomStore += it
                 }.map {
                     UserSubscriptionEvent.AddedToRoomEvent(it)
                 }
 
+                val updated = event.rooms.filter { nr ->
+                    knownRooms.any { kr ->
+                        kr.id == nr.id && (kr.name != nr.name || kr.customData != nr.customData)
```

Comment:
Perhaps the knowledge of what constitutes an identical `Room` object should be encapsulated in `Room`. If you redefine `equals` as suggested below, then a `deepEqual` or `allFieldsEqual` method could be added so this reads more like:

`if kr == nr && !kr.deepEquals(nr)`

Reply:
Yup! I was actually thinking about this on my walk in to work. Would've moved the logic into `Room`. Thanks!